hnn_core/tests/test_optimization.py

```diff
@@ -46,26 +47,84 @@ def test_consolidate_chunks():
 
 def test_split_by_evinput():
     """Test splitting evoked input."""
-    hnn_core_root = op.dirname(hnn_core.__file__)
-    params_fname = op.join(hnn_core_root, 'param', 'default.json')
-    params = read_params(params_fname)
+    drive_names = ['ev_drive_1', 'ev_drive_2']
+    drive_dynamics = [{'mu': 5., 'sigma': .1}, {'mu': 10., 'sigma': .2}]
+    drive_syn_weights = [{'ampa_L2_pyramidal': 1.}, {'nmda_L5_basket': 2.}]
+    tstop = 20.
+    dt = 0.025
 
     timing_range_multiplier = 3.0
     sigma_range_multiplier = 50.0
     synweight_range_multiplier = 500.0
     decay_multiplier = 1.6
-    evinput_params = _split_by_evinput(params, sigma_range_multiplier,
+    evinput_params = _split_by_evinput(drive_names, drive_dynamics,
+                                       drive_syn_weights, tstop,
+                                       sigma_range_multiplier,
                                        timing_range_multiplier,
                                        synweight_range_multiplier)
-    assert list(evinput_params.keys()) == [
-        'evprox_1', 'evdist_1', 'evprox_2']
+    assert list(evinput_params.keys()) == drive_names
     for evinput in evinput_params.values():
         assert list(evinput.keys()) == ['mean', 'sigma', 'ranges',
                                         'start', 'end']
 
-    evinput_params = _generate_weights(evinput_params, params,
+    evinput_params = _generate_weights(evinput_params, tstop, dt,
                                        decay_multiplier)
     for evinput in evinput_params.values():
         assert list(evinput.keys()) == ['ranges', 'start', 'end',
                                         'weights', 'opt_start',
                                         'opt_end']
+
+
+def test_optimize_evoked():
+    """Test running the full routine in a reduced network."""
+    hnn_core_root = op.dirname(hnn_core.__file__)
+    params_fname = op.join(hnn_core_root, 'param', 'default.json')
+    params = read_params(params_fname)
+
+    tstop = 10.
+    n_trials = 1
+
+    # simulate a dipole to establish ground-truth drive parameters
+    mu_orig = 6.
+    params.update({'N_pyr_x': 3,
+                   'N_pyr_y': 3,
+                   't_evprox_1': mu_orig,
+                   'sigma_t_evprox_1': 2.})
+    net_orig = jones_2009_model(params, add_drives_from_params=True)
+    del net_orig.external_drives['evprox2']
+    del net_orig.external_drives['evdist1']
+    dpl_orig = simulate_dipole(net_orig, tstop=tstop, n_trials=n_trials)[0]
+
+    # simulate a dipole with a time-shifted drive
+    mu_offset = 4.
+    params.update({'N_pyr_x': 3,
+                   'N_pyr_y': 3,
+                   't_evprox_1': mu_offset,
```

Comment:
```suggestion
                   't_evprox_1': mu_orig + mu_offset,
```
Was this the original intent?

Reply:
No, I was originally meaning for `mu_offset` to mean "the mu value that had been offset".